### DISCUSSION TITLE
Oauth scopes

### DISCUSSION THREAD

**Discussion by tigawanna** [pocketbase/pocketbase#1591] [Q&A] [answered]
Is there a way to include scopes in the oauth request, am trying to make something that'll let me replicate having a GitHub access token but with the GitHub oauth app , i already have the normal oauth working but with the default auth scope

**Comment by ganigeorgiev** [comment, marked as answer]
You should be able to change the default scopes by manually adjusting the `scope` query parameter in the provider login url that is returned with the "listAuthMethods()" response, eg. something like:

```js
const url = new URL(provider.authUrl + redirectUrl);

// replace the scope query param
// (by spec I think it should be space separated list, but refer to the provider docs if it doesn't work)
const params = url.searchParams;
params.set("scope", "myScope1 myScope2");
url.search = params.toString();

const newUrl = url.toString();
```

**Reply by tigawanna** [reply]
This looks easier than i thought it would be , thank you
